src/Arcus.Messaging.Pumps.Abstractions/Resiliency/DefaultMessagePumpCircuitBreaker.cs

```diff
@@ -44,25 +43,62 @@ public virtual async Task PauseMessageProcessingAsync(string jobId, Action<Messa
         {
             Guard.NotNullOrWhitespace(jobId, nameof(jobId));
 
-            _logger.LogTrace("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
-
             var options = new MessagePumpCircuitBreakerOptions();
             configureOptions?.Invoke(options);
 
             MessagePump messagePump = GetRegisteredMessagePump(jobId);
             if (messagePump.IsStarted)
             {
+                _logger.LogDebug("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
                 await messagePump.StopProcessingMessagesAsync(CancellationToken.None);
-                await WaitRecoveryTimeAsync(messagePump, options);
-                await TryProcessSingleMessageAsync(messagePump, options);
+
+                await Task.Factory.StartNew(async () =>
```

Comment:
I've been testing the CircuitBreaker functionality a bit more, and I have inconsistent results.
Sometimes the queue I'm processing gets completely processed, but more often then not, messages remain in the queue and the MessageProcessor is not retrieving anything anymore.

I suspect that this might have to do with these tasks that are started in the background.

Wouldn't it be an idea to change the 'strategy' of the message-receiver when the circuit-breaker transitions into a half-open state instead ?

Reply:
Let's park this for a while.